docs/content/docs/recipes/signals.md

```diff
@@ -0,0 +1,516 @@
+        /* [id]: DataObject */
+        signaler: Signaler,
+    },
+};
+```
+
+The `FluidContainer` is then created (or fetched if it already exists) alongside the `services` object, since audience will be required in this application. To learn more about how to create a new Fluid container and load an existing container, see our [DiceRoller Tutorial](https://fluidframework.com/docs/start/tutorial/).
+
+## MouseTracker
+
+Let's look at how the `MouseTracker` class uses the `Signaler` DataObject to achieve the first form of user presence within the application.
+
+The class defines the `mouseSignalType` that will be sent and listened to the connected clients when there is a presence change:
+
+```typescript
+export interface IMouseTrackerEvents extends IEvent {
+    (event: "mousePositionChanged", listener: () => void): void;
+}
+
+export class MouseTracker extends TypedEventEmitter<IMouseTrackerEvents> {
+    private static readonly mouseSignalType = "positionChanged";
+
+    /*...*/
+}
+```
+The class also initializes a local map of `IMousePosition` values for all of the connected clients. Each userID can have multiple clientIDs (e.g. same user on separate devices), which explains the nested `Map`. This local map is what populates the view and what will be updated on `mouseSignalType` signals:
+
+```typescript
+export interface IMouseTrackerEvents extends IEvent {
+    (event: "mousePositionChanged", listener: () => void): void;
+}
+
+export interface IMousePosition {
+    x: number;
+    y: number;
+}
+
+export class MouseTracker extends TypedEventEmitter<IMouseTrackerEvents> {
+    private static readonly mouseSignalType = "positionChanged";
+
+    /**
+     * Local map of mouse position status for clients
+     *
+     * ```
+     * Map<userId, Map<clientid, position>>
+     * ```
+     */
+    private readonly posMap = new Map<string, Map<string, IMousePosition>>();
+
+    /*...*/
+}
+```
+Now that we have clarity about what `posMap` and `mouseSignalType` are, we can move to the constructor to see what other behavior our `MouseTracker` has.
+
+We'll begin by looking at the constructor arguments. `MouseTracker` takes in the `audience` and the `Signaler` instance from `initialObjects`:
+
+```typescript
+export interface IMouseTrackerEvents extends IEvent {
+    (event: "mousePositionChanged", listener: () => void): void;
+}
+
+export class MouseTracker extends TypedEventEmitter<IMouseTrackerEvents> {
+    /*...*/
+
+    public constructor(
+      public readonly audience: IServiceAudience<IMember>,
+      private readonly signaler: Signaler,
+    ){
+      super();
+
+      /*...*/
+    }
+
+    /*...*/
+}
+```
+Whenever a member leaves the audience (e.g., a client disconnects from the container), we would want to remove their presence from our local data. After this is done, we would have to let the view know that the local presence data has changed. To do this, we emit our `mousePositionChanged` event so the view knows to re-render:
```

Comment:
Prior to talking about members leaving the audience, maybe talk about the `mousePositionChanged` event more generally -- e.g. "Whenever there is any change to the positions we're tracking, we'll fire a `mousePositionChanged` event to notify listeners that updated data is available."